src/utils/ConfigUtils.js

```diff
@@ -54,13 +28,127 @@ function getConversionMethod(param, subType, functionArray) {
+    'enumValues',
+    'hasHeader',
+    'hidden',
+    'subType',
+  ];
+  if (!knownAttributesNames.includes(attributeName)) {
+    console.warn(`The attribute "${attributeName}" is not a known attribute in the scenario parameters configuration.`);
+    return undefined;
+  }
+
+  // DEPRECATED: support for previous configuration path parameter.[attributeName] is deprecated and will be dropped in
+  // a future version
+  return parameter?.options?.[attributeName] ?? parameter?.[attributeName];
+};
+
+const getParametersGroupAttribute = (parametersGroup, attributeName) => {
+  const knownAttributesNames = ['authorizedRoles', 'hideParameterGroupIfNoPermission', 'hidden'];
+  if (!knownAttributesNames.includes(attributeName)) {
+    console.warn(
+      `The attribute "${attributeName}" is not a known attribute in the scenario parameters groups configuration.`
+    );
+    return undefined;
+  }
+
+  // DEPRECATED: support for previous configuration path parametersGroup.[attributeName] is deprecated and will be
+  // dropped in a future version
+  return parametersGroup?.options?.[attributeName] ?? parametersGroup?.[attributeName];
+};
+
+const _checkDeprecatedKeysInParametersGroupConfig = (parametersGroup) => {
+  const keysToCheck = ['authorizedRoles', 'hideParameterGroupIfNoPermission'];
+  keysToCheck.forEach((keyToCheck) => {
+    if (parametersGroup[keyToCheck] !== undefined) {
+      console.warn(
+        `The "parametersGroup.${keyToCheck}" key in the scenario parameters group configuration is now deprecated. ` +
+          `Please use "parametersGroup.options.${keyToCheck}" key instead.`
+      );
+    }
+  });
+};
+
+const checkDeprecatedKeysInConfig = (config) => {
+  config.parameters && config.parameters.forEach((parameter) => _checkDeprecatedKeysInParameterConfig(parameter));
+  config.parameterGroups &&
+    config.parameterGroups.forEach((parametersGroup) => _checkDeprecatedKeysInParametersGroupConfig(parametersGroup));
+};
+
+// Merge 2 arrays of elements based on their ids. If two elements with the same id have the same property, the property
+// value of the element in the merged array will be the property value from array2
+const mergeArraysByElementsIds = (array1, array2) => {
+  if (array1 == null) return array2;
+  if (array2 == null) return array1;
+  return array1
+    .map((el1) => ({
+      ...el1,
+      ...array2.find((el2) => el1.id === el2.id),
+    }))
+    .concat(array2.filter((el2) => !array1.some((el1) => el1.id === el2.id)));
+};
+
+const reshapeConfigArrayToDictById = (elementsArray) => {
+  const elementsDict = {};
+  for (const element of elementsArray ?? []) {
+    elementsDict[element.id] = element;
+  }
+  return elementsDict;
+};
+
+const reshapeDictToArrayById = (elementsDict) => {
+  const elementsArray = [];
+  for (const elementId in elementsDict) {
+    const element = elementsDict[elementId];
+    element.id = elementId;
+    elementsArray.push(element);
+  }
+  return elementsArray;
```

Comment:
We can create a ArrayDictUtils with generic function like 
ArrayToDict(array, IdPropertyName = 'id')
DictToArray(dict, idPropertyName = 'id)'
MergeArray(array1, array2, idPropertyName = 'id') ?